### PR TITLE
fix: enhance IPv6 host header handling in OAuth2 requests

### LINKED ISSUE #1022: Unhandled exception when Host header is IPv6 address
Running the service locally with a default configuration, then doing:

    curl "http://[::1]:8080/oauth/.well-known/openid-configuration"

## Expected behavior:

The service should respond with a `200 OK` and a response body like:

```json
{
  "issuer" : "http://[::1]:8080/oauth",
  "authorization_endpoint" : "http://[::1]:8080/oauth/authorize",
  "end_session_endpoint" : "http://[::1]:8080/oauth/endsession",
  "revocation_endpoint" : "http://[::1]:8080/oauth/revoke",
  "token_endpoint" : "http://[::1]:8080/oauth/token",
  "userinfo_endpoint" : "http://[::1]:8080/oauth/userinfo",
  "jwks_uri" : "http://[::1]:8080/oauth/jwks",
  "introspection_endpoint" : "http://[::1]:8080/oauth/introspect",
  "response_types_supported" : [ "code", "none", "id_token", "token" ],
  "response_modes_supported" : [ "query", "fragment", "form_post" ],
  "subject_types_supported" : [ "public" ],
  "id_token_signing_alg_values_supported" : [ "ES256", "ES384", "RS256", "RS384", "RS512", "PS256", "PS384", "PS512" ],
  "code_challenge_methods_supported" : [ "plain", "S256" ]
}
```

## Actual behavior:

    curl: (52) Empty reply from server

Logged exception:

```
2026-08-24 12:04:11 [nioEventLoopGroup-3-1] ERROR no.nav.security.mock.oauth2.http.OAuth2HttpServer - error in netty channel handler
java.lang.IllegalArgumentException: unexpected host: [
	at okhttp3.HttpUrl$Builder.host(HttpUrl.kt:993)
	at no.nav.security.mock.oauth2.http.OAuth2HttpRequest.proxyAwareUrl$mock_oauth2_server(OAuth2HttpRequest.kt:92)
	at no.nav.security.mock.oauth2.http.OAuth2HttpRequest.getUrl(OAuth2HttpRequest.kt:30)
	at no.nav.security.mock.oauth2.http.OAuth2HttpRouterKt$routeFromPathAndMethod$1.matchPath(OAuth2HttpRouter.kt:204)
	at no.nav.security.mock.oauth2.http.OAuth2HttpRouterKt$routeFromPathAndMethod$1.match(OAuth2HttpRouter.kt:207)
	at no.nav.security.mock.oauth2.http.PathRouter.findHandler(OAuth2HttpRouter.kt:126)
	at no.nav.security.mock.oauth2.http.PathRouter.invoke(OAuth2HttpRouter.kt:119)
	at no.nav.security.mock.oauth2.http.PathRouter.invoke(OAuth2HttpRouter.kt:108)
	at no.nav.security.mock.oauth2.http.NettyWrapper$RouterChannelHandler.channelRead0(OAuth2HttpServer.kt:237)
	at no.nav.security.mock.oauth2.http.NettyWrapper$RouterChannelHandler.channelRead0(OAuth2HttpServer.kt:210)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:107)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at io.netty.handler.codec.http.HttpServerKeepAliveHandler.channelRead(HttpServerKeepAliveHandler.java:64)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:361)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:325)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:171)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:796)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1474)
```

This is due to a bug in [java.lang.Thread.run(Thread.java:1474)](2026-08-24 12:04:11 [nioEventLoopGroup-3-1] ERROR no.nav.security.mock.oauth2.http.OAuth2HttpServer - error in netty channel handler
java.lang.IllegalArgumentException: unexpected host: [

### PR DESCRIPTION
**Pull request by ybelMekk**
This pull request improves how the server parses and handles HTTP `Host` headers, especially for IPv6 addresses and malformed input. The main changes include introducing a robust utility for splitting host and port from the `Host` header, updating server logic to use this utility, and expanding test coverage for edge cases.

**Host header parsing improvements:**

* Added a new internal function `hostAndPortFromHostHeader` that correctly splits host and port from the `Host` header, with special handling for bracketed IPv6 literals and malformed input. Ports outside the valid range now default to `-1` instead of causing errors. (`src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequest.kt`, [src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequest.ktR135-R155](diffhunk://#diff-3ab9cb664cfad177a80d548b8d7064094212eb2c7522d82a3540fe060a995179R135-R155))
* Updated `OAuth2HttpRequest` to use the new `hostAndPortFromHostHeader` utility, replacing the previous parsing logic. (`src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequest.kt`, [src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequest.ktL124-R125](diffhunk://#diff-3ab9cb664cfad177a80d548b8d7064094212eb2c7522d82a3540fe060a995179L124-R125))
* Refactored the Netty server (`NettyWrapper`) to use the new parsing function, ensuring only explicit, valid ports are used from the `Host` header. (`src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpServer.kt`, [src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpServer.ktL284-R290](diffhunk://#diff-499bbe9a985a79e509a505ae8bee1b7c633e7bd781f147939c61d7aa941349caL284-R290))

**Testing and validation:**

* Added and expanded tests to cover all `Host` header shapes, including various IPv6 formats, missing and malformed headers, and out-of-range ports. This ensures robust handling and prevents crashes or unexpected behavior. (`src/test/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequestTest.kt`, [src/test/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequestTest.ktR177-R223](diffhunk://#diff-af850122bf29f459870d6fb948b9cc47768e207e6eda87daadbebd57f906a73dR177-R223))
* Added integration tests to verify that requests with IPv6 `Host` headers are handled correctly and that the server does not crash or drop connections. (`src/test/kotlin/no/nav/security/mock/oauth2/e2e/WellKnownIntegrationTest.kt`, [[1]](diffhunk://#diff-7d30c14df28f95a521edc8eba09a2195f5d5725b58d912530912e6c6e6971103R46-R71); `src/test/kotlin/no/nav/security/mock/oauth2/server/OAuth2HttpServerTest.kt`, [[2]](diffhunk://#diff-a455cbef990fba837135650d89bbefbf2b2c2f7a149f8ebea818bb6dbcd692d8R85-R101)

These changes make the server more robust and proxy-aware, especially in environments using IPv6 or non-standard `Host` headers.

### CODE REVIEW

File: src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequest.kt [hunk 1]
```diff
@@ -139,3 +132,24 @@ data class OAuth2HttpRequest(
...
+    runCatching { URI("//$header") }.getOrNull()?.let { uri ->
+        if (uri.host != null) return uri.host to uri.port.asHostHeaderPort()
+    }
+
+    if (header.startsWith("[")) return null
+
+    val hostPort = header.split(":")
+    val port = if (hostPort.size == 2) hostPort[1].toIntOrNull()?.asHostHeaderPort() ?: -1 else -1
+    return hostPort[0] to port
+}
```
[the remark this answers is not in the document]
Comment (ybelMekk) [reply src/main/kotlin/no/nav/security/mock/oauth2/http/OAuth2HttpRequest.kt:153]: This case is already handled. `java.net.URI` does not range-check ports, so `URI("//[::1]:99999")` parses successfully with host `[::1]` and port `99999`, which `asHostHeaderPort()` then collapses to `-1`. The `startsWith("[")` bailout is only reached for genuinely unparseable literals (`[::1` unclosed, `[::1]:notaport` non-numeric), which a colon split can't salvage either. The existing tests cover this and pass: `hostAndPortFromHostHeader("[::1]:99999") shouldBe ("[::1]" to -1)`.
